web-components/src/merch-search.js

```diff
@@ -17,16 +17,31 @@ export class MerchSearch extends LitElement {
 
     constructor() {
         super();
-        this.handleInput = () =>
+
+        this.handleInput = () => {
+            pushStateFromComponent(this, this.search.value);
+        };
+        this.handleInputAndAnalytics = () => {
             pushStateFromComponent(this, this.search.value);
+            if (this.search.value) {
+                this.dispatchEvent(
+                    new CustomEvent('search-changed', {
```

Comment:
can you use a constant?

Reply:
Yes, now all events have the same name 
`export const EVENT_MERCH_CHANGE = 'merch-change';`
we can see in event.detail.type from where that event is thrown.